src/main/java/ru/testtask/service/FileService.java

```diff
@@ -0,0 +1,147 @@
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class FileService extends SimpleFileVisitor<Path> {
+
+    @Value("${upload.path}")
+    private String uploadPath;
+
+    private final FileDataRepo fileDataRepo;
+
+    private final MongoTemplate mongoTemplate;
+
+    public FileService(FileDataRepo fileDataRepo, MongoTemplate mongoTemplate) {
+        this.fileDataRepo = fileDataRepo;
+        this.mongoTemplate = mongoTemplate;
+    }
+
+    @PostConstruct
+    public void init(){
+        mongoTemplate.indexOps("files").ensureIndex(new Index("filename", Sort.Direction.ASC).unique());
+    }
+
+    public void addFile(Path path){
+        try {
+            FileData fileData = FileData.builder().originalFilename(path.toString())
+                    .filename(path.getFileName().toString())
+                    .size(Files.size(path))
+                    .type(Files.probeContentType(path))
+                    .directory(uploadPath)
+                    .build();
+
+            fileDataRepo.insert(fileData);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public List<Path> getFileListFromDirectory(Path path) throws IOException {
+
+        List<Path> paths = new ArrayList<>();
+
+        Files.walkFileTree(path, new SimpleFileVisitor<Path>() {
+            @Override
+            public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                if (!Files.isDirectory(file))
+                paths.add(file);
+                return super.visitFile(path, attrs);
+            }
+        });
+
+        return paths;
+    }
+
+    public void removeFile(String filename){
+        fileDataRepo.deleteByFilename(filename);
+    }
+
+    public FileData findByFilename(String filename) throws IOException {
+        Files.walkFileTree(Paths.get(uploadPath), new SimpleFileVisitor<Path>() {
+            @Override
+            public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                if (file.getFileName().toString().equals(filename))
+                    Files.delete(file);
+                return super.visitFile(file, attrs);
+            }
+        });
+        return fileDataRepo.findByFilename(filename);
+    }
+
+    public List<FileData> searchByRegex(String regex){
+        List<FileData> files = fileDataRepo.findAll();
+        List<FileData> appropriateFiles = new ArrayList<>();
+
+        for (FileData fileData : files){
+            if (fileData.getFilename().matches("(.*)" + regex + "(.*)"))
+                appropriateFiles.add(fileData);
+        }
+
+        return appropriateFiles;
+    }
+
+    public void moveFile(String filename, Path destination) throws IOException {
+
+        if (!Files.exists(destination)){
+            Files.createDirectory(destination);
+        }
+
+        Files.walkFileTree(Paths.get(uploadPath), new SimpleFileVisitor<Path>() {
```

Comment:
Before moving the file you need to check that destination dir doesn't contain file with similar file name. If file already exists, please return 409 status to client

Reply:
You will use file id instead of filename. So you will build current file path using it